### PR TITLE
Add filesize fetcher mode (need SOMEONE for testing)

### PR DESCRIPTION
**Pull request by ElegantMonkey**
Uses a torrent client to fetch the filesize of the torrents with `filesize IS NULL` or `filesize = 0` on the DB.

Seems to work, but needs to be tested with a more recent DB (especially with the HorribleSubs scraped one)

### CODE REVIEW

**Comment by tomleb** [comment]
Shouldn't that be a different application instead ?

**Comment by ElegantMonkey** [comment]
@tomleb: It is, kinda. You run it alongside the nyaa webapp, just like the scrapper, with `-mode filesize_fetcher`.

**Comment by majestrate** [comment]
I like this a lot

File: main.go [hunk 1]
```diff
@@ -142,6 +156,8 @@ func main() {
 			RunScraper(conf)
 		} else if *mode == "webapp" {
 			RunServer(conf)
+		} else if *mode == "filesize_fetcher" {
```
Comment (majestrate) [inline main.go]: nitpick: should be `metainfo_fetcher` instead as it fetches more than just the file sizes

Comment (ElegantMonkey) [inline main.go]: Yeah, but for now it only updates the filesize, so I think it's more appropriate calling it like this. But I can rename it if you want to.

**Comment by akuma06** [comment]
Just resolved your conflicts

**Comment by ElegantMonkey** [comment]
I would like to test on a recent DB really, only tested it with 1-2 torrents, as there are few recent torrents with no filesizes on the old one. Otherwise, I guess not.

**Comment by akuma06** [comment]
If you have tested it on 2 torrents, then it should be ok. I don't see why it would broke.

**Comment by ElegantMonkey** [comment]
Ok then, I hope it's fine.

**Comment by ewhal** [comment]
This is neat if we could also do something like this but for filelists then that would be really great!

**Comment by ElegantMonkey** [comment]
@ewhal, you mean the files inside the torrents? Probably could be done, but I don't know how we should store it (I don't think there's a field for it right now).

**Comment by ewhal** [comment]
We can always add a column to the db.
But yeah for the files in the torrent.

**Comment by ElegantMonkey** [comment]
@ewhal, I'll take a look, but could you test this with your DB? I've changed a few torrents manually to `filesize = 0` for testing, but for some of them I get 0 bytes as the length...

**Comment by ewhal** [comment]
sure will do and thank you for your help!

**Comment by ElegantMonkey** [comment]
Alright, now it reads the FileInfo from the metadata too. Probably should rename it, like @majestrate said, but I think it's good for now...

**Comment by ewhal** [comment]
Just tested it and it works

**Comment by ewhal** [comment]
@ElegantMonkey I didn't see that you had added filelist support and I just retested it and it doesn't work.
It successfully gets missing size information though just not the list of files per torrent.

**Pull request merged**